06.bowling_object/game.rb

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def calc_score(shot_scores)
+    frames = []
+    shot_scores.gsub('X', '10').split(',').each do |shot_score|
+      frames << Frame.new(frames.size) if frames.empty? || frames.last.finished?
+      frames.each do |frame|
+        frame.record_shot(shot_score.to_i)
+      end
+    end
+    frames.sum(&:frame_score)
+  end
+end
+
+puts Game.new.calc_score(ARGV[0])
```

Comment:
提出物のページでもらっていた質問の回答になりますが、こういうケースはクラスメソッドにするのがいいと思います。

```suggestion
puts Game.calc_score(ARGV[0])
```

トップレベルにあれこれ書いてしまうと、プロジェクト全体にそのコードが波及します。
使い捨てのスクリプトや、初心者用のお勉強プログラムならトップレベルでもいいですが、プロの現場では巨大なプログラムを書くので、極力トップレベルを汚染しないようなコードを書く必要があります。

この課題はかなりプラクティスとしてもかなり後半なので、プロの現場を意識してトップレベルを汚染しないコードを書いてもらうのが望ましいです。

Reply:
クラス化自体に意味があるかどうかというよりトップレベルを汚染しないためにクラス化させることもあるんですね。

`calc_score`をクラスメソッドに修正しました。